src/index.ts

```diff
@@ -108,15 +107,20 @@ export default class OpentracingExtension<TContext extends SpanContext>
       return;
     }
 
+    // indepotent method to add helpers to the first context available (which will be propagated by apollo)
+    addContextHelpers(context);
+
     const name = info.fieldName || "field";
-    const parentSpan = context.span ? context.span : this.requestSpan;
+    const parentSpan =
+      info.path && info.path.prev
+        ? context.getSpanByPath(info.path.prev)
+        : this.requestSpan;
 
     const span = this.localTracer.startSpan(name, {
       childOf: parentSpan || undefined
     });
 
-    // There is no nicer way to change the span in the context
-    context.span = span;
+    context.addSpan(span, info);
```

Comment:
Would you be willing to do the following?
```
info.span = span
```

I've taken @cliedeman's code and modified it to my needs, and this one addition was what I needed most.
It makes it super simple to add tracing to datasource-rest, and others by simply getting that property when I'm resolving my field.

Reply:
Interesting, I understand your use case and I think it's a common one. I would rather add and remove a field in the context, would that be okay for you? So you could do context.span.

Reply:
`context` is shared among all async resolvers, `info` is not. I had it in `context` in the beginning, and it kept getting overridden from various other resolvers, sadly. Perks of using async code, I guess.

Reply:
Fair enough, thank you for the explanation :) will add it to the info as soon as I am back in my room after graphql Finland

Reply:
Thanks a lot.
I'll be happy to test it when it gets in, and do some benchmarking on it.

Have a blast at the event! :D

Reply:
Great, thank you for helping :)